test/runner/e2e_update_rel_test.cpp

```diff
@@ -19,134 +19,372 @@ class UpdateRelTest : public DBTest {
+            // person0->person500(SET length=400)
+            *find(expectedResult.begin(), expectedResult.end(), to_string(3 * (100 - 1))) = "712";
+            *find(expectedResult.begin(), expectedResult.end(), to_string(3 * (500 - 1))) = "400";
+        }
+        auto result = conn->query("MATCH (:person)-[e:knows]->(p:person) return e.length");
+        auto actualResult = TestHelper::convertResultToString(*result);
+        sortAndCheckTestResults(actualResult, expectedResult);
     }
-    auto result = conn->query("MATCH (:person)-[e:knows]->(p:person) return e.length");
-    auto actualResult = TestHelper::convertResultToString(*result);
-    sortAndCheckTestResults(actualResult, expectedResult);
-}
-
-TEST_F(UpdateRelTest, UpdateNewlyInsertedEdges) {
-    conn->beginWriteTransaction();
-    ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("person" /* srcTable */, "person" /* dstTable */,
-        5 /*  srcID */, 8 /* dstID */, 10 /* lengthProp */)));
-    ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("person" /* srcTable */, "person" /* dstTable */,
-        7 /*  srcID */, 10 /* dstID */, 20 /* lengthProp */)));
-    ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("person" /* srcTable */, "person" /* dstTable */,
-        32 /*  srcID */, 51 /* dstID */, 30 /* lengthProp */)));
-    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
-        "knows" /* relation */, 5 /* srcID */, 8 /* dstID */, "set e.length = 55")));
-    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
-        "knows" /* relation */, 7 /* srcID */, 10 /* dstID */, "set e.length = null")));
-    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
-        "knows" /* relation */, 32 /* srcID */, 51 /* dstID */, "set e.length = 201")));
-    auto result =
-        conn->query("MATCH (p:person)-[e:knows]->(:person) where p.ID > 4 return e.length");
-    auto actualResult = TestHelper::convertResultToString(*result);
-    auto expectedResult = vector<string>{"", "55", "201"};
-    sortAndCheckTestResults(actualResult, expectedResult);
-}
-
-TEST_F(UpdateRelTest, UpdateRelsTwoHop) {
-    conn->beginWriteTransaction();
-    ASSERT_TRUE(conn->query("MATCH (a:animal)-[e1:knows]->(p:person)-[e2:knows]->(p1:person) WHERE "
-                            "p1.ID = 2500 AND a.ID = 5 SET e1.length = 300, e2.length = null"));
-    auto result =
-        conn->query("MATCH (a:animal)-[e1:knows]->(p:person)-[e2:knows]->(p1:person) WHERE "
-                    "p1.ID = 2500 AND a.ID = 5 RETURN e1.length, e2.length");
-    auto actualResult = TestHelper::convertResultToString(*result);
-    auto expectedResult = vector<string>{"300|"};
-    sortAndCheckTestResults(actualResult, expectedResult);
+
+    void updateNewlyAddedNodes(bool isCommit, TransactionTestType transactionTestType) {
```

Comment:
I'm a bit lost on this test. You are inserting new Knows rels but the function and the associated tests are called updateNewlyAdded*Nodes*. Why nodes? What's the difference with the updateNewlyInsertedRels function, where you add and update 3 rels (here you insert and update 2 rels).

Reply:
I think i am trying to:
1. insert rels to a newly added node in the current trx;
2. update those newly inserted rels.
The purpose of this testing is that: when doing commit, if the node is a newly added node, we will go through a special codepath.